cnxpublishing/views/admin.py

```diff
@@ -280,25 +280,30 @@ def admin_edit_site_message_POST(request):
              renderer='cnxpublishing.views:templates/print-style.html',
              permission='administer')
 def admin_print_styles(request):
+    """
+    Returns a dictionary of all unique print_styles, and their latest tag,
+    revision, and recipe_type.
+    """
     settings = request.registry.settings
     db_conn_str = settings[config.CONNECTION_STRING]
     styles = []
     with psycopg2.connect(db_conn_str) as db_conn:
         with db_conn.cursor() as cursor:
             cursor.execute("""\
-                SELECT print_style, fileid, recipe_type, revised, tag,
+                SELECT print_style, max(recipe_type),
+                    max(revised), max(tag),
```

Comment:
I don't think this is what you want.  This query is probably going to select the max `recipe_type`, max `revised` and max `tag` which might be from different rows in the database.

For example consider this table:

```
 print_style  | fileid | recipe_type | tag |            revised            
--------------+--------+-------------+-----+-------------------------------
 ccap-physics |      1 | web         | 1.0 | 2017-08-09 13:14:39.375236-07
 ccap-physics |      2 | pdf         | 2.0 | 2017-08-09 13:14:53.272308-07
```

The query is going to return:

```
 print_style  | max |              max              | max | count 
--------------+-----+-------------------------------+-----+-------
 ccap-physics | web | 2017-08-09 13:14:53.272308-07 | 2.0 |     0
```

I don't know if there's a more efficient way of doing this but this seems to work:

```sql
WITH latest AS (
    SELECT print_style, max(revised) AS revised
        FROM print_style_recipes
        GROUP BY print_style
) SELECT ps.print_style, ps.recipe_type, ps.revised, tag, (
    SELECT count(*)
        FROM latest_modules AS lm
        WHERE lm.print_style = ps.print_style AND lm.portal_type = 'Collection')
    FROM latest JOIN print_style_recipes AS ps ON ps.print_style = latest.print_style AND ps.revised = latest.revised;
```